### PR TITLE
restore pierce:15 to bionic energy weapons

### PR DESCRIPTION
**Pull request by swwu**
Bionic energy weapons used to have the same `pierce: 15` value that the handheld energy weapons do. It seems like in be78ad2e, this value was accidentally overlooked when the bionic weapon defs were moved from code into JSON. This PR should restore the pierce values to those weapons to bring them more in-line with comparable hand-held versions.

It's also worth noting that it seems most of the plasma weapons were upgraded from `["PLASMA", "INCENDIARY"]` to `["PLASMA", "EXPLOSIVE", "FLAME"]` while `bio_blaster` was left untouched. Would it make sense for `bio_blaster` to also get this upgrade?

(Of course, if the pierce change was an intended rebalance, feel free to close & ignore this PR)

### CODE REVIEW

**Comment by kevingranade** [comment]
I'll poke around a little bit, but I think you're right. Not adding the
flags to bio blaster also sounds like an oversight.

**Comment by swwu** [comment]
Updated with flag changes.

File: data/json/items/ranged.json [hunk 1]
```diff
@@ -38,7 +39,7 @@
     "material": ["steel", "plastic"],
     "skill": "rifle",
     "flags": ["NO_AMMO", "BIO_WEAPON", "NEVER_JAMS"],
-    "ammo_effects": ["PLASMA","INCENDIARY"],
+    "ammo_effects": ["PLASMA", "EXPLOSIVE", "FLAME", "NEVER_JAMS"],
```
Comment (Coolthulhu) [inline data/json/items/ranged.json]: "NEVER_JAMS" is already in flags. It doesn't do anything in ammo.

Comment (Rivet-the-Zombie) [inline data/json/items/ranged.json]: Ammo uses the NEVER_MISFIRES flag.

**Comment by swwu** [comment]
@Rivet-the-Zombie @Coolthulhu Ah, I just copied the line from the ftk93. Changed that one as well.

**Comment by swwu** [comment]
@kevingranade what results have your poking about gleaned?

**Pull request merged**